packages/babel-plugin/src/utils/traverse-expression-with-evaluation.tsx

```diff
@@ -0,0 +1,162 @@
+ *
+ * @param expression Expression we want to interrogate.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const traverseIdentifier = (expression: t.Identifier, meta: Metadata) => {
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  const binding = updatedMeta.parentPath.scope.getBinding(expression.name);
+  const resolvedBinding = resolveBindingNode(binding, updatedMeta);
+
+  if (resolvedBinding && resolvedBinding.constant) {
+    // We recursively call get interpolation until it not longer returns an identifier or member expression
+    ({ value, meta: updatedMeta } = traverseExpressionWithEvaluation(
+      resolvedBinding.node as t.Expression,
+      resolvedBinding.meta
+    ));
+  }
+
+  return createResultPair(value as t.Expression, updatedMeta);
+};
+
+/**
+ * Will look in an expression and return the actual value along with updated metadata.
+ *
+ * E.g: If there is a member expression called `colors.primary` that has identifier `color` which
+ * is set somewhere as `const colors = { primary: 'blue' }`,
+ * passing the `colors` identifier to this function would return `'blue'`.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const traverseMemberExpression = (expression: t.MemberExpression, meta: Metadata) => {
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  const { accessPath, bindingIdentifier } = getMemberExpressionMeta(expression);
+  const binding = updatedMeta.parentPath.scope.getBinding(bindingIdentifier.name);
+  const resolvedBinding = resolveBindingNode(binding, updatedMeta);
+
+  if (resolvedBinding && resolvedBinding.constant && t.isObjectExpression(resolvedBinding.node)) {
+    const objectValue = getValueFromObjectExpression(
+      resolvedBinding.node,
+      accessPath
+    ) as t.Expression;
+    // We recursively call get interpolation until it not longer returns an identifier or member expression
+    ({ value, meta: updatedMeta } = traverseExpressionWithEvaluation(
+      objectValue,
+      resolvedBinding.meta
+    ));
+  }
+
+  return createResultPair(value as t.Expression, updatedMeta);
+};
+
+/**
+ * Will look in an expression and return the actual value along with updated metadata.
+ *
+ * E.g: If there was a function called `size` that is set somewhere as
+ * `const size = () => 10` or `const size = function() { return 10; }` or `function size() { return 10; }`,
+ * passing the `size` identifier to this function would return `10` (it will recursively evaluate).
+ *
+ * @param expression Expression we want to interrogate.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const traverseFunction = (expression: t.Function, meta: Metadata) => {
```

Comment:
@madou: I can't get rid (letting runtime execute IIFE) of this traverse because we can have a function like this with an identifier `foo`

`const size = function() { return foo; }` which might be referencing `foo` from another module `const foo = 10;`

or may be a member expression 

`const size = function() { return foo.bar; }` which might be referencing `foo` from another module `const foo = { bar: 10 };`

So this will recursively find that and inline it. I also added a test for it.

Reply:
ok no worries i think this is fine